### PR TITLE
Misc clean up after recent PRs.

### LINKED ISSUE #795: Remove 'magic number' 3 from c4/Termination_Detector.cc
* Is 3 considered "magic number" code smell in this context, or is it okay?

_Originally posted by @keadyk in https://github.com/lanl/Draco/diffs_

* In `c4/Timer.cc`, eliminate multiple declarations/initialization found on the same line.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* This is a grab bag of tiny changes that I made locally while working on (1) reworking the coverage mechanics for interactive and regression systems and (2) a few changes that @keadyk recommended after reviewing #793.

### Purpose of Pull Request

* Fixes #795
* [Related to Redmine Issue #1903](https://rtt.lanl.gov/redmine/issues/1903)

### Description of changes

+ Small changes were implemented in `c4/Termination_Detector.cc` and `c4/Timer.cc` to address reviewer comments provided with PR #793 (clan-tidy fixes for c4).  This fixes #795.  Eliminate the magic number `3` from the first file. Ensure variable declarations are one per line for the second file.
+ `CTestCustom.cmake` was moved to the ccsradregress directory as it is only used by the CTest regression system.
+ Removed references to BullseyeCoverage as this capability is now depreciated.
  - Remove logic from `config/unix-g++.cmake` used to detect if Bullseye was enabled.
  - Remove a reference to `COVFILE` being set in `.bashrc`.
  - Remove comments about code that Bullseye reports as not covered.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,15 +103,6 @@ if( TARGET Exe_draco_info )
...
 
-#
-# Dashboard/Regression setup
-#
-# CTest/CDash custom configuration file
-configure_file(
-  ${PROJECT_SOURCE_DIR}/CTestCustom.cmake
-  ${PROJECT_BINARY_DIR}/CTestCustom.cmake
-  COPYONLY )
-
```
Comment (KineticTheory) [inline CMakeLists.txt:114]: No longer needed. Logic moved to `ccsradregress`.

File: config/unix-g++.cmake [hunk 2]
```diff
@@ -44,28 +44,8 @@ if( DEFINED CMAKE_CXX_COMPILER_ID )
...
-if( ${DBS_CXX_IS_BULLSEYE} MATCHES BullseyeCoverage )
-  set( DBS_CXX_IS_BULLSEYE "ON" )
-  set( DBS_CXX_IS_BULLSEYE ${DBS_CXX_IS_BULLSEYE}
-    CACHE BOOL "Are we running Bullseye" )
-  mark_as_advanced( DBS_CXX_IS_BULLSEYE )
-endif()
-
-# LTO (option '-flto') is controlled via the cmake variable
-# CMAKE_INTERPROCEDURAL_OPTIMIZATION, see config/compilerEnv.cmake.
-
```
Comment (KineticTheory) [inline config/unix-g++.cmake:68]: We no longer support bullseye coverage.

File: src/c4/Termination_Detector.cc [hunk 3]
```diff
@@ -65,7 +65,13 @@ Termination_Detector::~Termination_Detector() = default;
...
 //! Return \c true if we have terminated; \c false otherwise.
 bool Termination_Detector::is_terminated() {
-  static std::array<unsigned, 3> buffer;
+
+  // We keep track of three integral values in this buffer container.
+  // - send_count
+  // - receive_count
+  // - work_count
+  size_t constexpr num_counts = 3;
+  std::array<unsigned, num_counts> buffer;
```
Comment (KineticTheory) [inline src/c4/Termination_Detector.cc:74]: Eliminate use of magic number `3`.

File: src/c4/Timer.cc [hunk 4]
```diff
@@ -340,11 +327,14 @@ void Timer::printline_mean(std::ostream &out, unsigned const p,
...
-  double s = sum_system_cpu(), s2 = s * s;
-  double ww = sum_wall_clock(), ww2 = ww * ww;
+  double ni = num_intervals;
+  double ni2 = ni * ni;
+  double u = sum_user_cpu();
+  double u2 = u * u;
+  double s = sum_system_cpu();
+  double s2 = s * s;
+  double ww = sum_wall_clock();
+  double ww2 = ww * ww;
```
Comment (KineticTheory) [inline src/c4/Timer.cc:337]: One variable declaration per line.

**Comment by KineticTheory** [comment]
@keadyk This is also ready for final review.  It addresses some of the items you noticed in on of my recent PRs plus a couple of additional issues.

**Comment by keadyk** [comment]
LGTM, thanks! Merging now :)

**Pull request merged**